libs/config/src/asio.cpp

```diff
@@ -0,0 +1,8 @@
+//  Copyright (c) 2019 Thomas Heller
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/config/asio.hpp>
+
+// This file is intentionally empty to ensure self contained headers
```

Comment:
What are these source files needed for?

Reply:
Why do we need explicit source files for this? Those could be generated (as before), couldn't they?

Reply:
The biggest issue with the autogenerated header tests is that you easily end up with dangling headers. This completely eliminates this issue at the cost of added verbosity.

Reply:
Please elaborate, I'm not sure I understand what you mean. The generated files look exactly as the files you created manually (**shiver**), well - perhaps sans the license comment.

Reply:
@sithhell could you explain how we might end up with dangling headers (unused headers, I presume?) more easily if the tests are auto-generated?